hyperspy/io_plugins/ripple.py

```diff
@@ -521,16 +532,11 @@ def file_writer(filename, signal, encoding='latin-1', *args, **kwds):
     # Check if the dtype is supported
     dc = signal.data
     dtype_name = signal.data.dtype.name
+    dimension = len(signal.data.shape)
     if dtype_name not in dtype2keys.keys():
         err = 'The ripple format does not support writting data of %s type' % (
             dtype_name)
         raise IOError(err)
-    # Check if the dimensions are supported
```

Comment:
Yes, you are right. However in the future we may split the plugins into a separate project so that other packages can import them without having to import the whole HyperSpy and this error may be handy if this ever happens.